cmd/container-disk-v2alpha/main.c

```diff
@@ -0,0 +1,142 @@
+#include <ctype.h>
+#include <getopt.h>
+#include <libgen.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include <sys/socket.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/un.h>
+#include <unistd.h>
+
+#define LISTEN_BACKLOG 50
+#define READINESS_PROBE_FILE "/healthy"
+
+static void *socket_check(void *arg) {
+    struct stat st = {0};
+    char *copy_path;
+    copy_path = (char *)arg;
+
+    for (;;) {
+        sleep(1);
+        if (stat(copy_path, &st) == -1) {
```

Comment:
This is going to cause constant (albeit small) load on the system. Would it be possible to use something along the lines of inotify or select to block instead of constantly polling?

Reply:
I just re-used code from the https://github.com/kubevirt/kubevirt/blob/a6aa0981230f97b9996f57f73691e7376b9fddf1/cmd/container-disk-v2alpha/main.go#L78

I do not have the intention to change the code under this PR, but we can leave it for the following PR.